app/controllers/search_controller.rb

```diff
@@ -2,8 +2,15 @@ class SearchController < ApplicationController
   skip_before_filter :authenticate_user!, :only => [ :result ]
   def result
     # First search in morphology mode, if fails — retry in wildcard mode
-    @projects = Project.search params[:q]
-    @projects = Project.search params[:q], star: true if @projects.empty?
+    episode = params[:episode]
+
+    if episode != "all"
+      @projects = Project.search params[:q],conditions: {episode_names: episode}
```

Comment:
`params[:episode]` is  `Episode.id` and not `Episode.name` 

Would be easy to get the name if you would use `@episode` which is the object.